backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/outputGenerators/DocxGenerator.java

```diff
@@ -293,13 +294,16 @@ private void fillOverview3rdPartyComponentTable(XWPFDocument document, Collectio
             row.addNewTableCell().setText(licenseInfo.getSha1Hash());
             row.addNewTableCell().setText(licenseInfo.getComponentName());
 
-            String globalLicense = "";
-            for(LicenseNameWithText l : licenseInfo.getLicenseNamesWithTexts()) {
-                if(l != null && l.getType().equals("global")) {
-                    globalLicense = l.getLicenseName();
-                    break;
-                }
-            }
+            Optional<String> opGlobalLicense = licenseInfo.getLicenseNamesWithTexts().stream()
```

Comment:
What about `if(l != null && "global".equals(l.getType()) {`
The root cause is just the NPE of `.getType`.
I think your implementation is working quite good but somehow not really readable.

Reply:
A null pointer check is not very elegant. But of course your suggestion is more readable.